### PR TITLE
Added onError prop

### PR DESCRIPTION
**Pull request by generalChaos**

### CODE REVIEW

File: src/Form.js [hunk 1]
```diff
@@ -9,12 +9,14 @@ export default class Form extends Component {
   static defaultProps = {
     defaultErrors: undefined,
     defaultValue: undefined,
+    onValidationError: undefined,
```
Comment (alansouzati) [inline src/Form.js]: can we call this onError?

File: src/Form.js [hunk 2]
```diff
@@ -55,13 +57,18 @@ export default class Form extends Component {
...
-    const { onSubmit } = this.props;
+    const { onSubmit, onValidationError } = this.props;
     const { formState, resource } = this.state;
+    let formErrors = {};
     event.preventDefault();
     if (formState.isValid()) {
       onSubmit(resource);
     } else {
-      this.setState({ isFormSubmitted: true, errors: formState.getErrors() });
+      formErrors = formState.getErrors();
```
Comment (alansouzati) [inline src/Form.js]: can this be `const errors = formState.getErrors();` ?

File: src/Form.js [hunk 3]
```diff
@@ -55,13 +57,18 @@ export default class Form extends Component {
...
+      if (typeof onValidationError === 'function') {
+        onValidationError(formErrors);
+      }
+      this.setState({ isFormSubmitted: true, errors: formErrors });
```
Comment (alansouzati) [inline src/Form.js]: I believe this can be simplified to be `this.setState({ isFormSubmitted: true, errors });`

**Comment by alansouzati** [comment]
hey @generalChaos thanks so much for your contribution and sorry for taking such a long time to review this. I've added few minor comments, if you can apply that changes it would be great.

**Comment by generalChaos** [comment]
@alansouzati   No problem, I took care of the changes.  Let me know if there are any other changes.  Thanks for considering this PR.

**Comment by alansouzati** [comment]
you are the best, thanks so much for your contribution 🎉 🍺

**Comment by alansouzati** [comment]
release 0.8.0 with this feature include.

**Pull request merged**